lib/libzfs/libzfs_pool.c

```diff
@@ -4536,6 +4587,23 @@ zpool_obj_to_path(zpool_handle_t *zhp, uint64_t dsobj, uint64_t obj,
 	free(mntpnt);
 }
 
+/*
+ * Given an dataset object number, return data block and indirect block size.
+ */
+void
+zpool_get_block_size(zpool_handle_t *zhp, uint64_t obj,
```

Comment:
GIven that this function returns void, it should always succeed; if the ioctl returns an error, right now we just won't touch the data_blk_size or indrt_blk_size, which may not be the ideal behavior.  It should probably return an error to inform the caller it couldn't get the sizes, and let the caller deal with that as they choose.